### PR TITLE
Tooling rehaul: introduce ARM64 release; use git tags instead of `production` branch for releases

### PR DESCRIPTION
**Pull request by peterdeme**
# Description of the change

Tooling rehaul.

## Multi-arch uploads to S3 (downloads.spacelift.io)
This pull request makes it so that we'll upload platform-specific binaries to S3 **and** multi-platform Docker images as well.

How it used to look like:
- https://downloads.spacelift.io/spacelift-vcs-agent
- https://downloads.spacelift.io/spacelift-vcs-agent_SHA256SUMS
- https://downloads.spacelift.io/spacelift-vcs-agent_SHA256SUMS.sig

☝️  Let's call this legacy. 

New format:
- https://downloads.spacelift.io/spacelift-vcs-agent-x86_64
- https://downloads.spacelift.io/spacelift-vcs-agent-x86_64_SHA256SUMS
- https://downloads.spacelift.io/spacelift-vcs-agent-x86_64_SHA256SUMS.sig
- https://downloads.spacelift.io/spacelift-vcs-agent-aarch64
- https://downloads.spacelift.io/spacelift-vcs-agent-aarch64_SHA256SUMS
- https://downloads.spacelift.io/spacelift-vcs-agent-aarch64_SHA256SUMS.sig

This naming convention entirely matches the launcher binaries.
❗ **We will still upload the legacy files**, but we shouldn't use those links anywhere. I'll update the documentation at docs.spacelift.io once this PR is merged.

## New release process

A new use-case arose in Spacelift where we'd like to download VCS Agent binaries for Darwin for end-to-end testing purposes. CC @jakubdal 
In order to facilitate that, let's improve our release process: instead of merging to the `production` branch, releases should happen when a new tag is pushed. This follows our other public repositories' pattern such as spacectl or terraform-provider-spacelift.

From now on, Goreleaser will publish a GitHub release whenever a tag is pushed and it'll attach all the built binaries as well.

How does that help us? We will publish all 3 operating systems (Linux/Darwin/Windows) to the GitHub Releases which can be used by endtoend tests - those tests can detect the current OS and architecture and download the agent accordingly.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue);
- [x] New feature (non-breaking change that adds functionality);
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected);
- [ ] Other (miscellaneous, GitHub workflow changes, changes to the PR template);

### Development

- [x] Lint rules pass locally;
- [x] All tests related to the changed code pass in development;

### Code review

- [ ] This pull request has a descriptive title and sufficient context for a reviewer. There may be a screenshot or screencast attached;
- [ ] This pull request is no longer a draft;

### Deployment

- [x] Selected merge strategy is [squash merge](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits);
- [ ] Changes have been reviewed by at least one other engineer;

### CODE REVIEW

**Review by eminalemdar — APPROVED** [review]
LGTM

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -2,9 +2,13 @@ name: Linting
...
 on: { push: { branches-ignore: [main, production] } }
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
 jobs:
   linting:
-    name: Lint the code
+    name: 🤓 Lint the code
```
Comment (jakubdal) [inline .github/workflows/linting.yml:11]: ❤️

**Comment by peterdeme** [comment]
@jakubdal ahh there was a typo, fixed it, do you mind reapproving? 🙏

**Review by jakubdal — APPROVED** [review]

File: .github/workflows/publish/action.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        cp dist/spacelift-vcs-agent_linux_arm64/spacelift-vcs-agent_SHA256SUMS.sig build/spacelift-vcs-agent-aarch64_SHA256SUMS.sig
+    
+        # For visibility, here's the list of files we're uploading:
+        ls -l build
+    
+    - name: Upload the VCS Agent binaries to downloads.spacelift.[dev|io]
+      if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/') }}
+      shell: bash
+      run: aws s3 sync build/ s3://${{ inputs.aws_bucket }}
+
```
Comment (eminalemdar) [inline .github/workflows/publish/action.yml:108]: Probably I'm missing something here but where is container image build and push action? @peterdeme

Comment (peterdeme) [reply .github/workflows/publish/action.yml:108]: goreleaser @eminalemdar

Comment (eminalemdar) [reply .github/workflows/publish/action.yml:108]: thank you! all clear now!

**Review by eminalemdar — APPROVED** [review]
LGTM!

**Pull request merged**